src/main/java/com/puppycrawl/tools/checkstyle/SuppressionsStringPrinter.java

```diff
@@ -0,0 +1,99 @@
+import com.puppycrawl.tools.checkstyle.xpath.XpathQueryGenerator;
+
+/**
+ * Class for constructing xpath queries to suppress nodes
+ * with specified line and column number.
+ * @author Timur Tibeyev.
+ */
+public final class SuppressionsStringPrinter {
+
+    /** Line and column number config value pattern. */
+    private static final Pattern VALID_SUPPRESSION_LINE_COLUMN_NUMBER_REGEX =
+            Pattern.compile("^([0-9]+):([0-9]+)$");
+
+    /** Prevent instances. */
+    private SuppressionsStringPrinter() {
+        // no code
+    }
+
+    /**
+     * Prints generated suppressions.
+     * @param file the file to process.
+     * @param suppressionLineColumnNumber line and column number of the suppression
+     * @param tabWidth length of the tab character
+     * @return generated suppressions.
+     * @throws IOException if the file could not be read.
+     * @throws CheckstyleException if the file is not a Java source.
+     */
+    public static String printSuppressions(File file, String suppressionLineColumnNumber,
+                                           int tabWidth) throws IOException, CheckstyleException {
+        final Matcher matcher =
+                VALID_SUPPRESSION_LINE_COLUMN_NUMBER_REGEX.matcher(suppressionLineColumnNumber);
+        if (matcher.matches()) {
+            final FileText fileText = new FileText(file.getAbsoluteFile(),
+                    System.getProperty("file.encoding", StandardCharsets.UTF_8.name()));
+            final DetailAST detailAST =
+                    JavaParser.parseFileText(fileText, JavaParser.Options.WITH_COMMENTS);
+            final int lineNumber = Integer.parseInt(matcher.group(1));
+            final int columnNumber = Integer.parseInt(matcher.group(2));
+            return generate(fileText, detailAST, lineNumber, columnNumber, tabWidth);
+        }
+        else {
+            final String exceptionMsg = String.format(Locale.ROOT,
+                    "%s does not match valid format 'line:column'.",
+                    suppressionLineColumnNumber);
+            throw new IllegalStateException(exceptionMsg);
+        }
+    }
+
+    /**
+     * Creates {@code XpathQueryGenerator} instance and generates suppressions.
+     * @param fileText {@code FileText} object.
+     * @param detailAST {@code DetailAST} object.
+     * @param lineNumber line number.
+     * @param columnNumber column number.
+     * @param tabWidth length of the tab character.
+     * @return generated suppressions.
+     */
+    private static String generate(FileText fileText, DetailAST detailAST, int lineNumber,
+                                   int columnNumber, int tabWidth) {
+        final XpathQueryGenerator queryGenerator =
+                new XpathQueryGenerator(detailAST, lineNumber, columnNumber, fileText,
+                        tabWidth);
+        final List<String> suppressions = queryGenerator.generate();
+        return suppressions.stream().collect(Collectors.joining("\n"));
```

Comment:
use System.lineSeparator() instead of "\n" where possible

Reply:
Should It print "new line" character in the end?